staging/src/k8s.io/client-go/dynamic/fake/simple_test.go

```diff
@@ -303,3 +304,139 @@ func TestPatch(t *testing.T) {
 		t.Run(tc.name, tc.runner)
 	}
 }
+
+// This test ensures list works when the fake dynamic client is seeded with a typed scheme and
+// unstructured type fixtures
+func TestListWithUnstructuredObjectsAndTypedScheme(t *testing.T) {
```

Comment:
I'd like to see a test case where the list isn't seeded at all and the response needs to be fully constructed.

Reply:
Added `TestListWithNoFixturesAndTypedScheme`

I'm testing with a typed scheme - otherwise I'd expect consumers to use the other constructor that requires a mapping of gvrs to list kinds